Source/common/BUILD

```diff
@@ -425,6 +425,7 @@ objc_library(
         ":SNTRule",
         ":SNTStrengthify",
         ":SNTSystemInfo",
+        "//Source/santasyncservice:Pinning",
```

Comment:
We shouldn't bring anything outside of common into common.

We shouldn't need to modify the configurator for this change. What if we just changed the `IsDomainPinned` function to be named something more straightforward - `DomainSupportsSyncV2` - and then in `SNTSyncManager`, we can just check:

```
} else if (config.enableNATS == YES && santa::DomainSupportsSyncV2(config.syncBaseURL)) {
```